### DISCUSSION TITLE
Persist in nested arrays

### DISCUSSION THREAD

**Discussion by timkolar** [alpinejs/alpine#2233] [1. Help] [answered]
I can't get Persist to work in a nested array.

It works like this:

```
Alpine.data("shop", function () {
  return {
    products: this.$persist([
      {
        id: 0,
        label: "Mleto 100g",
        price: 1.2,
        tax: 9.5,
        weight: 100,
        qty: 0, 
        image: "img/mleto-min.jpg"
      },...
```

But I need it to work like this:

```
Alpine.data("shop", function () {
  return {
    products: [
      {
        id: 0,
        label: "Mleto 100g",
        price: 1.2,
        tax: 9.5,
        weight: 100,
        qty: this.$persist(0), 
        image: "img/mleto-min.jpg"
      },...
```

Thank you in advance.

**Comment by SimoTod** [comment, marked as answer]
Nested object are supported but object nested in array aren't, right now.
Even if the support was added, it wouldn't be very robust because array keys are kind of random and they are not really related to the object. You should probably persist the whole list or pick a different data structure (maybe use an object keyed by ID).
